### ISSUE TITLE
Punycoded routes written with non-ascii characters fails silently

### ISSUE THREAD

**Issue by straend** [traefik/traefik#7494]
**Bug**

Punycodes in router written with non ascii characters.
  
Will generate valid Lets Encrypt certificates but won't match when routing

### What did you do?
With the following 

```yml
version: "3"
networks:
  webb:
    external: true
services:
  nonpuny:
    image: traefik/whoami
    restart: unless-stopped
    networks:
      - webb
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=webb"
      - "traefik.http.routers.npun.rule=Host(`nonpuny.example.com`)"
      - "traefik.http.routers.npun.entrypoints=websecure"
      - "traefik.http.routers.npun.tls.certresolver=myresolver"
      
  punycode:
    image: traefik/whoami
    restart: unless-stopped
    networks:
      - webb
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=webb"
      - "traefik.http.routers.pun.rule=Host(`punycöde.example.com`)"
      - "traefik.http.routers.pun.entrypoints=websecure"
      - "traefik.http.routers.pun.tls.certresolver=myresolver"
  punycoded:
    image: traefik/whoami
    restart: unless-stopped
    networks:
      - webb
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=webb"
      - "traefik.http.routers.pun2.rule=Host(`xn--punycded-r4a.example.com`)"
      - "traefik.http.routers.pun2.entrypoints=websecure"
      - "traefik.http.routers.pun2.tls.certresolver=myresolver"
```

`nonpuny` and `punycoded` works as intended

`punycode` will get a certificate indicating that everything works but will generate a 404 not found in Traefik when accessed

The first and last one will work as intended. Second one gets certificate but responds with `404 page not found` 

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
Version:      2.3.2
Codename:     picodon
Go version:   go1.15.3
Built:        2020-10-19T18:36:22Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
```
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.web.address=:80"
      - "--entrypoints.websecure.address=:443"
      - "--certificatesresolvers.myresolver.acme.httpchallenge=true"
      - "--certificatesresolvers.myresolver.acme.httpchallenge.entrypoint=web"
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

No related info shows up.

**Comment by rtribotte** [comment]
Hello @straend,

Thanks for your interest in Traefik!

It seems that you'll never reach Traefik with the `Host` header matching you rule.
I assume that your are looking for the support of unicode hostnames for the `Host` rule, by punicoding it.
So I label it as an enhancement.

**Comment by traefiker** [comment]
Closed by #7986.
